src/sentry/debug_files/artifact_bundles.py

```diff
@@ -150,7 +150,10 @@ def _index_urls_in_bundle(
             return
 
         # Insert the index
-        ArtifactBundleIndex.objects.bulk_create(urls_to_index)
+        # NOTE: The django ORM by default tries to batch *all* the inserts into a single query,
+        # which is not quite that efficient. We want to have a fixed batch size,
+        # which will result in a fixed number of unique `INSERT` queries.
+        ArtifactBundleIndex.objects.bulk_create(urls_to_index, batch_size=50)
```

Comment:
Is there any way to determine a good batch size?

Reply:
There has to be some kind of sweet spot. I’m actually surprised django did not define a default. The way I found this was that it creates *a ton* of unique queries which I bet is bad for metrics purposes. Both for datadog and for sentry performance span metrics.
So optimizing for cardinality, the number should be fairly small. But optimizing for less queries, the number should be higher.
Actually generating the raw SQL string, and parsing that on the server side is also not trivial, which might also indicate that optimizing for less cardinality is a good idea.